File: API-Server/Happy Habits App/Forms/ToiletForm.cs
using Happy_Habits_App.Model;
using System.Text.Json.Serialization;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace Happy_Habits_App.Forms
{
    public class ToiletForm : HabitForm
    {
        [JsonPropertyName("type")]
        public string? Type { get; set; } = null;
        [JsonPropertyName("time")]
        public string? Hour { get; set; } = null;
        [JsonPropertyName("notes")]
        public string? Note { get; set; } = null;

        public bool IsValid
        {
            get
            {
                return !string.IsNullOrEmpty(UserId) &&
                       !string.IsNullOrEmpty(Date) &&
                       !string.IsNullOrEmpty(Type) &&
                       !string.IsNullOrEmpty(Hour) &&
                       (Note != null);
            }
        }
    }
}

File: Mobile-App/app/src/main/java/com/example/happyhabits/feature_application/feature_toilet/data/model/ToiletForm.kt
package com.example.happyhabits.feature_application.feature_toilet.data.model

import kotlinx.serialization.Serializable

@Serializable
data class ToiletForm (
    val userId: String,
    val date: String,
    val type: String,
    val time: String,
    val notes: String
) {}
